test/e2e/mv3/multiple-restarts.spec.js

```diff
@@ -0,0 +1,294 @@
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  openDapp,
+  generateGanacheOptions,
+  WALLET_PASSWORD,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+const WINDOW_TITLES = Object.freeze({
+  ExtensionInFullScreenView: 'MetaMask',
+  TestDApp: 'E2E Test Dapp',
+  Notification: 'MetaMask Notification',
+  ServiceWorkerSettings: 'Inspect with Chrome Developer Tools',
+  InstalledExtensions: 'Extensions',
+});
+
+describe('MV3 - Restart service worker multiple times', function () {
+  it('Simple simple send flow within full screen view should still be usable', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: generateGanacheOptions(),
+        title: this.test.title,
+        driverOptions: { openDevToolsForTabs: true },
+      },
+      async ({ driver, ganacheServer }) => {
+        await driver.navigate();
+
+        await unlockWallet(driver, WALLET_PASSWORD);
+
+        await assertETHBalance(ganacheServer, '25');
+
+        console.log('first send ETH and then terminate SW');
+        const RECIPIENT_ADDRESS = '0x985c30949c92df7a0bd42e0f3e3d539ece98db24';
+        await simpleSendETH(driver, '1', RECIPIENT_ADDRESS);
+        await terminateServiceWorker(driver);
+
+        await assertETHBalance(ganacheServer, '24.0000');
+
+        console.log('first send ETH #2 and then terminate SW');
+        await switchToWindow(driver, WINDOW_TITLES.ExtensionInFullScreenView);
+        await simpleSendETH(driver, '1', RECIPIENT_ADDRESS);
+        await terminateServiceWorker(driver);
+
+        await assertETHBalance(ganacheServer, '22.9999');
```

Comment:
Question: would the gas fee different everytime and result in different balance?